packages/core/src/geometry/BufferSystem.ts

```diff
@@ -0,0 +1,149 @@
+import { System } from '../System';
+import { GLBuffer } from './GLBuffer';
+
+import type { Renderer } from '../Renderer';
+import type { IRenderingContext } from '../IRenderingContext';
+import type { Buffer } from './Buffer';
+
+/**
+ * System plugin to the renderer to manage buffers.
+ *
+ * @class
+ * @extends PIXI.System
+ * @memberof PIXI.systems
+ */
+export class BufferSystem extends System
+{
+    CONTEXT_UID: number;
+    gl: IRenderingContext;
+
+    readonly managedBuffers: {[key: number]: Buffer};
+
+    /**
+     * @param {PIXI.Renderer} renderer - The renderer this System works for.
+     */
+    constructor(renderer: Renderer)
+    {
+        super(renderer);
+
+        /**
+         * Cache for all buffers by id, used in case renderer gets destroyed or for profiling
+         * @member {object}
+         * @readonly
+         */
+        this.managedBuffers = {};
+    }
+
+    /**
+     * Sets up the renderer context and necessary buffers.
+     */
+    protected contextChange(): void
+    {
+        this.gl = this.renderer.gl;
+
+        // TODO fill out...
+        this.CONTEXT_UID = this.renderer.CONTEXT_UID;
+    }
+
+    /**
+     * this binds specified buffer. OOn first run, it will create the webGL buffers for the context too
+     *
+     * @param {PIXI.Buffer} buffer the buffer to bind to the renderer
+     */
+    bind(buffer:Buffer):void
+    {
+        const { gl, CONTEXT_UID } = this;
+
+        const glBuffer = buffer._glBuffers[CONTEXT_UID] || this.createGLBuffer(buffer);
+
+        // TODO can cache this on buffer! maybe added a getter / setter?
+        const type = buffer.index ? gl.ELEMENT_ARRAY_BUFFER : gl.ARRAY_BUFFER;
+
+        gl.bindBuffer(type, glBuffer.buffer);
+    }
+
+    /**
+     * Will ensure sure the the data in the buffer is uploaded to the GPU.
+     *
+     * @param {PIXI.Buffer} buffer the buffer to update
+     */
+    update(buffer:Buffer): void
+    {
+        const { gl, CONTEXT_UID } = this;
+
+        const glBuffer = buffer._glBuffers[CONTEXT_UID];
+
+        if (buffer._updateID !== glBuffer.updateID)
+        {
+            glBuffer.updateID = buffer._updateID;
```

Comment:
If there's no "else" may be use same approach as in https://github.com/pixijs/pixi.js/blob/v5.1.x/packages/core/src/framebuffer/FramebufferSystem.js#L439 , just return out of function in if.